contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,426 @@
+     * @param amount Amount of ETH to deposit into Trove
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+     * @param getId Optional storage slot to retrieve the ETH from
+    */
+    function deposit(
+        uint amount,
+        address upperHint,
+        address lowerHint,
+        uint getId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam)  {
+        if (getId != 0 && amount != 0) {
+            revert("deposit(): Cannot supply an amount if a non-zero getId is supplied");
+        }
+        amount = getUint(getId, amount);
+        borrowerOperations.addColl{value: amount}(upperHint, lowerHint);
+        _eventName = "LogDeposit(address,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, amount, getId);
+    }
+
+    /**
+     * @dev Withdraw ETH from Trove
+     * @notice Move Trove collateral from Trove to DSA
+     * @param amount Amount of ETH to move from Trove to DSA
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+     * @param setId Optional storage slot to store the withdrawn ETH in
+    */
+   function withdraw(
+        uint amount,
+        address upperHint,
+        address lowerHint,
+        uint setId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam)  {
+        borrowerOperations.withdrawColl(amount, upperHint, lowerHint);
+
+        setUint(setId, amount);
+        _eventName = "LogWithdraw(address,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, amount, setId);
+    }
+    
+    /**
+     * @dev Mints LUSD tokens
+     * @notice Borrow LUSD via an existing Trove
+     * @param maxFeePercentage The maximum borrow fee that this transaction should permit 
+     * @param amount Amount of LUSD to borrow
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+     * @param setId Optional storage slot to store the borrowed LUSD in
+    */
+    function borrow(
+        uint maxFeePercentage,
+        uint amount,
+        address upperHint,
+        address lowerHint,
+        uint setId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam)  {
+        borrowerOperations.withdrawLUSD(maxFeePercentage, amount, upperHint, lowerHint);
+
+        setUint(setId, amount); // TODO: apply fee / get exact amount borrowed (with the fee applied)
```

Comment:
Is this still TODO or is it OK to store the drawn LUSD amount?

Reply:
Yeah the drawn LUSD amount is the value we want to share with future spells